pkg/apis/planetscale/v2/vitessshard_defaults.go

```diff
@@ -16,10 +16,24 @@ limitations under the License.
 
 package v2
 
+import (
+	"k8s.io/utils/pointer"
+)
+
 // DefaultVitessShard fills in VitessShard defaults for unspecified fields.
 // Note: This should only be used for nillable fields passed down from a parent because controllers run in parallel,
 // and the defaulting code for a parent object may not have been run yet, meaning the values passed down from that parent
 // might not be safe to deref.
 func DefaultVitessShard(dst *VitessShard) {
 	DefaultTopoReconcileConfig(&dst.Spec.TopologyReconciliation)
+	defaultInitializeMaster(&dst.Spec.VitessShardTemplate.Replication.InitializeMaster)
 }
+
+func defaultInitializeMaster(enabled *bool) {
+	// Enable initialization of replication by default.
+	enableInitMaster := enabled
+
+	if enableInitMaster == nil {
```

Comment:
Also, now that you made the type non-pointer this will never be nil. You are always taking a valid address when you call this defaulting function, so now it doesn't ever default.

Reply:
Fixed.